onecodex/lib/upload.py

```diff
@@ -482,13 +481,15 @@ def _s3_intermediate_upload(file_obj, file_name, fields, session, callback_url):
 
     # issue a callback
     try:
+        # retry on 502, 503, 429, with a backoff timing of 4s, 8s, and 16s
         resp = session.post(
             callback_url,
             json={
                 "s3_path": "s3://{}/{}".format(fields["s3_bucket"], fields["file_id"]),
                 "filename": file_name,  #
                 "import_as_document": fields.get("import_as_document", False),
             },
+            max_retries=Retry(total=3, backoff_factor=4, status_forcelist=[502, 503, 429]),
```

Comment:
I believe `method_whitelist` needs to explicitly be specified as POSTs aren't considered idempotent. We _probably_ want to retry on a generic 500 too, though that really shouldn't be happening (in general, I'd avoid this as we don't know it's idempotent, but in this case the server-side operation _is_).